internal/keycloakb/passwordgeneration.go

```diff
@@ -0,0 +1,100 @@
+package keycloakb
+
+import (
+	"math/rand"
+	"strconv"
+	"strings"
+	"time"
+	"unicode"
+)
+
+// GeneratePassword generates a password of a given length
+func GeneratePassword(minLength int) string {
+
+	var pwdElems []string
+	const alphabet = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ?!*0123456789"
+
+	for j := 0; j < minLength; j++ {
+		pwdElems = append(pwdElems, string(alphabet[rand.Intn(len(alphabet))]))
+	}
+
+	pwd := strings.Join(pwdElems, "")
+	return pwd
+}
+
+// GeneratePasswordFromKeycloakPolicy generates a random password respecting the keycloak password policy
+func GeneratePasswordFromKeycloakPolicy(policy string, minLength int) (string, error) {
+	// Keycloak password policy is a string of the form
+	// "forceExpiredPasswordChange(365) and specialChars(1) and upperCase(1) and lowerCase(1) and length(4) and digits(1) and notUsername(undefined)"
+	var pwdElems []string
+	policyItems := strings.Split(policy, "and")
+
+	// generate a random password that corresponds to the password policy
+	//reg := regexp.MustCompile(`[a-zA-z]+[(]{1}[0-9]+[)]{1}`)
+	//pwdReq := string(reg.Find([]byte()))
+
+	f := func(c rune) bool {
+		return !unicode.IsLetter(c) && !unicode.IsNumber(c)
+	}
+
+	for i := 0; i < len(policyItems); i++ {
```

Comment:
We could first parse the policy to get a map[string]int (only with parameters having a number between parenthesis).
Then we could call a method appendCharacters like this:

> appendCharacters(pwdElems, "abcdefghijklmnopqrstuvwxyz", map, "length")
> appendCharacters(pwdElems, "0123456789", map, "digits")
> appendCharacters(pwdElems, "?!#%$", map, "specialChars")